### PR TITLE
[Issue #3080] Error handling in the login endpoints

### LINKED ISSUE #3080: Error handling in the login endpoints
### Summary

When an error occurs we want to redirect to the same final location that a user goes to when they successfully login, except with some additional error details.

Rough notes:
* Putting a try/except around all of the logic in the endpoint is probably simplest, although consider a decorator for re-usability.
* If the error is one of a few common known ones (like our own jwt errors) then we should pass through its message, otherwise put a generic "Internal Server Error" style message
* The logic for redirect/building the URL from its pieces will already exist and just needs to be re-used

### Acceptance criteria

- [ ] Error handling added
- [ ] Tests added to validate the behavior for various cases

### PR DESCRIPTION
**Pull request by chouinar**
## Summary
Fixes #3080

## Changes proposed
Error handling added to the login endpoints using a decorator that is keeps us from needing duplicate work

## Context for reviewers
Normally putting a big `try/catch` around an entire API endpoint is a bad idea, we should let errors propagate, but if an error occurs (lets say the user has an issue in login.gov), we need the error to go somewhere - because these endpoints are all redirect endpoints, we send them to the "final" endpoint where we want to put the error.

If you aren't familiar with the decorator added here, it is just a wrapper around the function, effectively it lets us re-use a bit of code and "wrap" a function, in this case we effectively do:

```py
try:
     our_api_func(...)
except ...:
     ...
```

## Additional information
There aren't any actual places that can error without deliberately breaking things right now - so I've overriden random methods to break - as we build out the callback endpoint, plenty of errors will go through this logic and we'll change it a bit more.

### CODE REVIEW

File: api/src/auth/auth_utils.py [hunk 1]
```diff
@@ -1,8 +1,76 @@
...
+        @functools.wraps(f)
+        def wrapper(*args: P.args, **kwargs: P.kwargs) -> flask.Response:
+            try:
+                return f(*args, **kwargs)
+            except HTTPError as e:
+                # HTTPError is what raise_flask_error raises
+                # and should encompass our "expected" errors
+                # that aren't a concern, as long as it isn't a 5xx
+                message = e.message
+                logger.info("Login flow failed")
```
Comment (mdragon) [inline api/src/auth/auth_utils.py:54]: Do we want to include the message here so it's clearer what logs here correspond to what users are experiencing?

**Review by mdragon — DISMISSED** [review]

File: api/src/auth/auth_utils.py (hunk 1 above)
Comment (chouinar) [reply api/src/auth/auth_utils.py:54]: I think I realized that about two PRs after this one and adjusted some of the logging here, I'll backport it after I merge the upstream changes

**Comment by mdragon** [comment]
I approved it, but you noted there would be later churn, so feel free to tag me back in.

**Comment by chouinar** [comment]
> I approved it, but you noted there would be later churn, so feel free to tag me back in.

@mdragon - https://github.com/HHS/simpler-grants-gov/pull/3087 needs to be approved first, these auth PRs are chained together so I need to get them approved/merged one-by-one

**Review by mdragon — APPROVED** [review]

**Pull request merged**